code/modules/awaymissions/mission_code/UO71-terrorspiders.dm

```diff
@@ -191,26 +191,28 @@
 	origin_tech = null
 	selfcharge = 1
 	can_charge = 0
-	var/inawaymission = 1
+	var/inawaymission = TRUE
 
-/obj/item/gun/energy/laser/awaymission_aeg/process()
+/obj/item/gun/energy/laser/awaymission_aeg/Initialize(mapload)
+	RegisterSignal(src, list(COMSIG_MOVABLE_Z_CHANGED), .proc/updateZ)
+	// Force update it incase it spawns outside an away mission and shouldnt be charged
+	updateZ()
+	return ..()
```

Comment:
Shouldn't this happen before the above, the `..()`? Usually Initiations are handled base -> self. And destroys self -> base due to how dependencies can work. This rule can be broken if you need to override something that gets used in the init of a base.
It should be fine here though but just for consistence.